src/messages/MOSDOp.h

```diff
@@ -358,11 +358,11 @@ struct ceph_osd_request_head {
 
     // Always keep here the newest version of decoding order/rule
     if (header.version == HEAD_VERSION) {
-	  ::decode(pgid, p);
-	  ::decode(osdmap_epoch, p);
-	  ::decode(flags, p);
-	  ::decode(reassert_version, p);
-	  ::decode(reqid, p);
+      ::decode(pgid, p);
```

Comment:
or we can reorder this part so it is consistent with `encode_payload()`? we need to keep sync with it whenever we bump up the `HEAD_VERSION`, the logic of `encode_payload()` looks like:
```c
if (!(feature & CEPH_FEATURE_OBJECTLOCATOR)) {
  header.version = 1;
  // ....
} else if (!(features & CEPH_FEATURE_NEW_OSDOP_ENCODING)) {
  header.version = 6;
  // ...
} else {
  header.version = HEAD_VERSION;
  // ...
}
```

so speaking of consistency, we can put something like
```c
if (header.version < 2) {
  // ...
} else if (header.version < HEAD_VERSION) {
  // ...
} else {
  // ...
}
```

what do you think?